### PR TITLE
Dynamic price corretions/changes

### PR DESCRIPTION
**Pull request by IonAgorria**
Dynamic price module corrections and changes

With this the already deprecated method Product->setPriceExpression is unused, should be deleted? I will commit that if so

### CODE REVIEW

**Comment by eldy** [comment]
We can keep the deprecated function more time.

File: htdocs/product/dynamic_price/class/price_parser.class.php [hunk 1]
```diff
@@ -131,6 +131,15 @@ public function translatedError()
 	public function parseExpression($product, $expression, $values)
 	{
 		global $user;
+
+		//Check if empty
+		$expression = trim($expression);
+		if (empty($expression))
+		{
+			$this->error = array(20, null);
```
Comment (eldy) [inline htdocs/product/dynamic_price/class/price_parser.class.php:139]: What is this ?
this->error shoud contains a string (one error message string or code)
this->errors can contains an array of different error messages

Comment (IonAgorria) [inline htdocs/product/dynamic_price/class/price_parser.class.php:139]: Price Parser uses error var for passing a numeric error and extra values, it was always this way https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/product/dynamic_price/class/price_parser.class.php

**Comment by eldy** [comment]
Hum, not compliant with rest of software.
In a future, if you can rename the var into  errorparser  for example to avoid conflict with ->error and ->errors used by framework.

**Comment by IonAgorria** [comment]
I will rename it as soon as posible.

**Pull request merged**